ospf6d/ospf6_route.c

```diff
@@ -1168,10 +1168,12 @@ void ospf6_route_show_detail(struct vty *vty, struct ospf6_route *route)
 	/* Nexthops */
 	vty_out(vty, "Nexthop:\n");
 	for (ALL_LIST_ELEMENTS_RO(route->nh_list, node, nh)) {
+		struct interface *ifp;
 		/* nexthop */
 		inet_ntop(AF_INET6, &nh->address, nexthop, sizeof(nexthop));
-		ifname = ifindex2ifname(nh->ifindex, ospf6->vrf_id);
-		vty_out(vty, "  %s %.*s\n", nexthop, IFNAMSIZ, ifname);
+
+		ifp = if_lookup_by_index_all_vrf(nh->ifindex);
+		vty_out(vty, "  %s %.*s\n", nexthop, IFNAMSIZ, ifp->name);
```

Comment:
Nit: this `nexthop`/`inet_ntop` could be replaced with a `%pI6` format option in `vty_out`.

E.g.

```c
vty_out(vty, "  %pI6 %.*s\n", &nh->address, IFNAMSIZ, ifp->name);
```

Reply:
Resolved.

Reply:
Resolved